extensions/agent/src/mainController.ts

```diff
@@ -91,6 +97,36 @@ export class MainController {
 			}
 			this.proxyDialog.dialogName ? await this.proxyDialog.openDialog(this.proxyDialog.dialogName) : await this.proxyDialog.openDialog();
 		});
+		vscode.commands.registerCommand('agent.openNotebookEditorFromJsonString', async (filename: string, jsonNotebook: string) => {
+			const tempfilePath = path.join(os.tmpdir(), filename + '.ipynb');
+			if (fs.existsSync(tempfilePath)) {
```

Comment:
try and avoid sync function calls. you can use promisfy from the utils library to use the non sync version as async promisee `import { promisfy } from 'utils'`